app/controllers/api/base_controller/renderer.rb

```diff
@@ -549,6 +549,18 @@ def render_options(resource, data = {})
         render :json => OptionsSerializer.new(klass, data).serialize
       end
 
+      # TODO: not sure if we should keep a separate update
+      #       or have 2 copy/paste methods
+      #       changing all params_for_update seems a bit much
+      def render_resource_options(id, action = :update)
+        type = @req.collection.to_sym
+        resource = resource_search(id, type)
+        raise BadRequestError, resource.unsupported_reason(action) unless resource.supports?(action)
+
+        schema = action == :update ? resource.params_for_update : resource.params_for_action(action)
```

Comment:
> resource.params_for_action(action)

This is going to end up with a method and a giant `case/when` statement in the providers, might it be better to have a `params_for_#{action}` method?

Reply:
we can do this. was hoping to avoid the dynamic method generation but you are probably right

Reply:
Yeah 100% know what you mean, it is guarded by the `resource.supports?(action)` so we aren't just `.send()`'ing anything, and we discussed checking the collection actions from api.yml that should mitigate any user provided input issues